src/System.Linq.Expressions/src/System/Linq/Expressions/Interpreter/LightCompiler.cs

```diff
@@ -1215,6 +1215,7 @@ private void EmitUnaryMethodCall(UnaryExpression node)
                 _instructions.EmitBranch(computed);
 
```

Comment:
I think the code needs  _instructions.EmitPop(); after _instructions.EmitBranchFalse(notNull);

It may also be useful to use EmitCoalescingBranchhere similarly to how it is used in CompileCoalesceBinaryExpression. It is essentially a coalescing branch.